drivers/sensor/ntc_thermistor/ntc_thermistor.c

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright (c) 2023 Google LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/drivers/adc.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/logging/log.h>
+#include "ntc_thermistor.h"
+
+LOG_MODULE_REGISTER(NTC_THERMISTOR, CONFIG_SENSOR_LOG_LEVEL);
+
+struct ntc_thermistor_data {
+	struct k_mutex mutex;
+	int16_t raw;
+	int16_t sample_val;
+};
+
+struct ntc_thermistor_config {
+	const struct adc_dt_spec adc_channel;
+	const struct ntc_config ntc_cfg;
+};
+
+static const struct ntc_compensation comp_epcos_b57861s0103a039[] = {
+	{ -25, 146676 },
+	{ -15, 78875 },
+	{ -5, 44424 },
+	{ 5, 26075 },
+	{ 15, 15881 },
+	{ 25, 10000 },
+	{ 35, 6488 },
+	{ 45, 4326 },
+	{ 55, 2956 },
+	{ 65, 2066 },
+	{ 75, 1474 },
+	{ 85, 1072 },
+	{ 95, 793 },
+	{ 105, 596 },
+	{ 115, 454 },
+	{ 125, 351 },
+};
```

Comment:
Maybe move this to where you specify the compatible below?
```c
#undef DT_DRV_COMPAT
#define DT_DRV_COMPAT epcos_b57861s0103a039
DT_INST_FOREACH_STATUS_OKAY_VARGS(NTC_THERMISTOR_DEV_INIT, DT_DRV_COMPAT)
```

Reply:
done thanks